rsciio/arina/_api.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2024 The HyperSpy developers
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with any project and source this library is coupled.
+# If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import logging
+from pathlib import Path
+
+import h5py
+import hdf5plugin  # noqa F401
+import numpy as np
+
+_logger = logging.getLogger(__name__)
+
+
+def file_reader(
+    filename,
+    lazy=False,
+    scan_width=None,
```

Comment:
We try to enforce consistent parameters name between reader and the `scan_width` would be similar to `navigation_shape`. If you want to define only one dimension, maybe something like `navigation_shape=(100, "auto")` can be used?